lib/parity/environment.rb

```diff
@@ -37,8 +41,14 @@ def backup
     end
 
     def deploy
-      if deploy_to_heroku && run_migration?
+      pending_migrations = run_migration?
+
+      deployed = deploy_to_heroku
```

Comment:
Yeah, I'm really not wild about this code as it stands. As another alternative, we could save state. Your suggestion is an improvement over what's there, I'll implement it.